utils/auth.js

```diff
@@ -0,0 +1,34 @@
+const jwt = require('jsonwebtoken');
+const { promisify } = require('util');
+const { getUserById } = require('../models/userDao');
+
+const loginRequired = async (req, res, next) => {
+
+  // 1) Getting token and check if it is there
+  const accessToken = req.headers.authorization;
+  if (!accessToken) {
+    const error = new Error('NEED_ACCESS_TOKEN');
+    error.statusCode = 401;
+
+    return res.status(error.statusCode).json({ message : error.message });
+  }
+
+  // 2) Verification token
+  const decoded = await promisify(jwt.verify)(accessToken, process.env.JWT_SECRET);
+  console.log(decoded)
+  // 3) Check if user still exists
+  const user = await getUserById(decoded.userId);
+
+  if(!user) {
+    const error = new Error('USER_DOES_NOT_EXIST');
+    error.statusCode = 404;
+
+    return res.status(error.statusCode).json({ message : error.message });
+  }
+
+  // 4) Grant access
+  req.user = user;
```

Comment:
DAO에서 호출 된 user를 반환하지 말고 토큰 자체에서 반환하고 있는 payload 정보에서 유저 정보를 반환해주세요!
```suggestion
  req.user = decoded;
  // 혹은
  req.user = docoded.id // payload에 담은 유저의 정보 활용
```